core/src/env.rs

```diff
@@ -184,22 +186,90 @@ impl<E: Ext> LaterExt<E> {
 
         res
     }
+}
+
+/// Struct for interacting with Ext.
+///
+/// Unlike [`BaseExtCarrier`] this struct is intended for external usage.
+pub struct ExtCarrier<E: Ext>(BaseExtCarrier<E>);
+
+impl<E: Ext> ExtCarrier<E> {
+    /// New ext carrier
+    pub fn new(e: E) -> Self {
+        Self(BaseExtCarrier::new(e))
+    }
+
+    /// Unwraps hidden `E` value
```

Comment:
It's a good practice to mention when function panics

Reply:
Actually we have proved that it doesn't panic. Do we still need mentioning this?

Reply:
Oh I forgot about `qed`